cylc/flow/workflow_files.py

```diff
@@ -1331,9 +1331,9 @@ def validate_workflow_name(name: str) -> None:
             "Workflow name cannot be a path that points to the cylc-run "
             "directory or above"
         )
-    if re.findall('run(N|[0-9]+)', Path(name).name):
+    if runNcheck and re.findall(r'^run(N|\d+)$', Path(name).name):
```

Comment:
We should ban it anywhere in the name (e.g. `bart/runN/lisa` should be banned)
```suggestion
    if runNcheck and any(
        re.match(r'^run(N|\d+)$', dir_name)
        for dir_name in Path(name).parts
    ):
```